docs/DafnyRef/integration-cs/IntegrationCS.md

```diff
@@ -0,0 +1,172 @@
+---
+title: Integration Dafny with C# projects
+---
+
+The Dafny compilation process translates Dafny programs into target language
+source code. 
+
+For a simple Dafny-only program, the translation step converts a `A.dfy` file into `A.cs`;
+the build step then produces a `A.dll`, which can be used as a library or as an executable.
+
+A multi-language program that combines Dafny and C#
+code "just" needs to be sure that the translated Dafny code fits in
+to the C# code. There are two aspects to this:
+- ensuring that the names of entities in the translated Dafny code are usable from C#
+- ensuring that the types are the same on both sides
+
+## Calling C# from Dafny
+
+Calling a C# method from Dafny requires declaring a shim in Dafny that gives a name and types
```

Comment:
```suggestion
Calling a C# method from Dafny requires declaring a [shim](https://en.wikipedia.org/wiki/Shim_(computing)) in Dafny that gives a name and types
```
I had never heard of the term "shim" before, so I think it's appropriate to define it on the fly.